instrumentation/apache-camel-2.20/javaagent/src/test/groovy/test/SqsCamelTest.groovy

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test
+
+import static io.opentelemetry.api.trace.Span.Kind.CLIENT
+import static io.opentelemetry.api.trace.Span.Kind.CONSUMER
+import static io.opentelemetry.api.trace.Span.Kind.INTERNAL
+import static io.opentelemetry.api.trace.Span.Kind.PRODUCER
+
+import com.google.common.collect.ImmutableMap
+import io.opentelemetry.instrumentation.test.AgentTestRunner
+import io.opentelemetry.instrumentation.test.utils.PortUtils
+import org.apache.camel.CamelContext
+import org.apache.camel.ProducerTemplate
+import org.elasticmq.rest.sqs.SQSRestServer
+import org.elasticmq.rest.sqs.SQSRestServerBuilder
+import org.springframework.boot.SpringApplication
+import org.springframework.context.ConfigurableApplicationContext
+import spock.lang.Shared
+
+class SqsCamelTest extends AgentTestRunner {
+
+  @Shared
+  ConfigurableApplicationContext server
+
+  @Shared
+  SQSRestServer sqs
+  @Shared
+  int sqsPort
+
+  def setupSpec() {
+    sqsPort = PortUtils.randomOpenPort()
+    sqs = SQSRestServerBuilder.withPort(sqsPort).withInterface("localhost").start()
+    println getClass().name + " SQS server started at: localhost:$sqsPort/"
+
+    def app = new SpringApplication(SqsConfig)
+    app.setDefaultProperties(ImmutableMap.of("sqs.port", sqsPort))
+    server = app.run()
+  }
+
+  def cleanupSpec() {
+    if (server != null) {
+      server.close()
+      server = null
+    }
+    if (sqs != null) {
+      sqs.stopAndWait()
+    }
+  }
+
+  def "simple sqs producer-consumer services"() {
+    setup:
+    def camelContext = server.getBean(CamelContext)
+    ProducerTemplate template = camelContext.createProducerTemplate()
+
+    when:
+    template.sendBody("direct:input", "{\"type\": \"hello\"}")
+    // wait for SQS to send back
+    Thread.sleep(2000)
```

Comment:
Instead of waiting consider adding a processor calling `CyclicBarrier#await()` in the SQS consumer route - a similar thing is done in `JaxRsHttpServerTest`.

Reply:
I think that right now it's not even needed (local SQS).